nixos/modules/security/pam.nix

```diff
@@ -1458,6 +1477,16 @@ in
       }
     ];
 
+    warnings = optional
+      (with lib; with config.security.pam.sshAgentAuth;
+        enable -> any (s: hasPrefix "%h" s || hasPrefix "~" s) authorizedKeysFiles)
```

Comment:
`->` should be changed to `&&` to fix eval warning.

Reply:
If `enable` is false and `authorizedKeyFiles` is default, then this condition is true, which will trigger the warning.